autotest/ogr/ogr_ngw.py

```diff
@@ -0,0 +1,495 @@
+        return 'skip'
+
+    if check_availability(gdaltest.ngw_test_server) == False:
+        gdaltest.ngw_drv = None
+        return 'skip'
+
+    ds_resource_id = gdaltest.ngw_ds.GetMetadataItem('id', '')
+    gdaltest.ngw_ds = None
+
+    url = 'NGW:' + gdaltest.ngw_test_server + '/resource/' + ds_resource_id
+    gdaltest.ngw_ds = gdal.OpenEx(url, gdal.OF_UPDATE, open_options=['BATCH_SIZE=2'])
+
+    lyr = gdaltest.ngw_ds.GetLayerByName('test_pt_layer')
+    f1 = ogr.Feature(lyr.GetLayerDefn())
+    fill_fields(f1)
+    f1.SetGeometry(ogr.CreateGeometryFromWkt('POINT (1 2)'))
+    ret = lyr.CreateFeature(f1)
+    if ret != 0 or f1.GetFID() >= 0:
+        return 'fail'
+
+    f2 = ogr.Feature(lyr.GetLayerDefn())
+    fill_fields2(f2)
+    f2.SetGeometry(ogr.CreateGeometryFromWkt('POINT (2 3)'))
+    ret = lyr.CreateFeature(f2)
+    if ret != 0 or f2.GetFID() >= 0:
+        return 'fail'
+
+    f3 = ogr.Feature(lyr.GetLayerDefn())
+    fill_fields(f3)
+    f3.SetGeometry(ogr.CreateGeometryFromWkt('POINT (3 4)'))
+    ret = lyr.CreateFeature(f3)
+    if ret != 0:
+        return 'fail'
+
+    ret = lyr.SyncToDisk()
+    if ret != 0:
+        return 'fail'
+
+    lyr.ResetReading()
+    feat = lyr.GetNextFeature()
+    counter = 0
+    while feat is not None:
+        counter += 1
+        if feat.GetFID() < 0:
+            gdaltest.post_reason('Expected FID greater or equal 0, got {}.'.format(feat.GetFID()))
+            return 'fail'
+
+        feat = lyr.GetNextFeature()
+
+    if counter < 3:
+        gdaltest.post_reason('Expected 3 or greater feature count, got {}.'.format(counter))
+        return 'fail'
+
+    return 'success'
+
+###############################################################################
+# Check ExecuteSQL.
+
+def ogr_ngw_9():
+    if gdaltest.ngw_drv is None:
+        return 'skip'
+
+    if check_availability(gdaltest.ngw_test_server) == False:
+        gdaltest.ngw_drv = None
+        return 'skip'
+
+    gdaltest.ngw_ds.ExecuteSQL('DELLAYER:test_ln_layer')
+    lyr = gdaltest.ngw_ds.GetLayerByName('test_ln_layer')
+    if lyr is not None:
+        gdaltest.post_reason('Expected fail to get layer test_ln_layer.')
+        return 'fail'
+
+    lyr = gdaltest.ngw_ds.GetLayerByName('test_pl_layer')
+
+    f = ogr.Feature(lyr.GetLayerDefn())
+    fill_fields(f)
+    f.SetGeometry(ogr.CreateGeometryFromWkt('POLYGON((0 0,0 1,1 0,0 0))'))
+    ret = lyr.CreateFeature(f)
+    if ret != 0:
+        gdaltest.post_reason('Failed to create feature in test_ln_layer.')
+        return 'fail'
+
+    if lyr.GetFeatureCount() != 1:
+        gdaltest.post_reason('Expected feature count is 1, got {}.'.format(lyr.GetFeatureCount()))
+        return 'fail'
+
+    gdaltest.ngw_ds.ExecuteSQL('DELETE FROM test_pl_layer;')
+    if lyr.GetFeatureCount() != 0:
+        gdaltest.post_reason('Expected feature count is 0, got {}.'.format(lyr.GetFeatureCount()))
+        return 'fail'
+
+    return 'success'
+
```

Comment:
would be interesting to run test_ogrsf

Reply:
```bash
./test_ogrsf NGW:http://dev.nextgis.com/sandbox/resource/0
INFO: Open of `NGW:http://dev.nextgis.com/sandbox/resource/0' using driver `NGW' successful.
INFO: Internal data source name `Main resource group'
      different from user name `NGW:http://dev.nextgis.com/sandbox/resource/0'.
INFO: Testing layer test_mvt.
INFO: Feature count verified.
INFO: Feature/layer spatial ref. consistency verified.
INFO: Spatial filter inclusion seems to work.
INFO: Spatial filter exclusion seems to work.
INFO: Spatial filter is ignored by GetFeature() as expected.
INFO: Infinity spatial filter works as expected.
INFO: Huge coords spatial filter works as expected.
INFO: Attribute filter inclusion seems to work.
INFO: Attribute filter exclusion seems to work.
INFO: Attribute filter is ignored by GetFeature() as expected.
INFO: GetExtent() test passed.
INFO: Random read test passed.
INFO: SetNextByIndex() read test passed.
INFO: Delete Feature test passed.
INFO: Feature was created, but with not its original FID.
INFO: Create Feature test passed.
INFO: Random write test passed.
INFO: Layer has UTF-8 content and is consistently declared as having UTF-8 content.
INFO: Transactions test skipped due to lack of transaction support.
INFO: TestLayerSQL passed.
```

Reply:
Cool, but I meant putting it into a test like done in other drivers

Reply:
Fixed per 12da9726242ce149b5e0abb5614ea2ec363fe952